// This file is automatically generated by sila2lib.proto_builder.proto_builder version 0.3.4
// :generation date: 2021-03-04 18:18:34
//
// ---- PLEASE DO NOT MODIFY MANUALLY !! ---

syntax = "proto3";
import "SiLAFramework.proto";
package sila2.de.fau.dispensing.ot2controller.v1;

// Feature: OT-2 Controller
//   A SiLA 2 complaint controller for an OT-2 liquid handler.
service Ot2Controller {
    // Upload Protocol
    //   Uploads the given Protocol to the "/data/user_storage" directory on the OT-2.
    rpc UploadProtocol(sila2.de.fau.dispensing.ot2controller.v1.UploadProtocol_Parameters) returns (sila2.de.fau.dispensing.ot2controller.v1.UploadProtocol_Responses) {}
    // Remove Protocol
    //   Removes the given Protocol from the "/data/user_storage" directory on the OT-2.
    rpc RemoveProtocol(sila2.de.fau.dispensing.ot2controller.v1.RemoveProtocol_Parameters) returns (sila2.de.fau.dispensing.ot2controller.v1.RemoveProtocol_Responses) {}
    // Run Protocol
    //   Runs the given Protocol on the OT-2.
    rpc RunProtocol(sila2.de.fau.dispensing.ot2controller.v1.RunProtocol_Parameters) returns (sila2.de.fau.dispensing.ot2controller.v1.RunProtocol_Responses) {}
    // Connection
    //   Connection details of the remote OT-2.
    rpc Get_Connection(sila2.de.fau.dispensing.ot2controller.v1.Get_Connection_Parameters) returns (sila2.de.fau.dispensing.ot2controller.v1.Get_Connection_Responses) {}
    // Available Protocols
    //   List of the stored files available on the OT-2.
    rpc Get_AvailableProtocols(sila2.de.fau.dispensing.ot2controller.v1.Get_AvailableProtocols_Parameters) returns (sila2.de.fau.dispensing.ot2controller.v1.Get_AvailableProtocols_Responses) {}
    // Camera Picture
    //   A current picture from the inside of the OT-2 made with the built-in camera.
    rpc Get_CameraPicture(sila2.de.fau.dispensing.ot2controller.v1.Get_CameraPicture_Parameters) returns (sila2.de.fau.dispensing.ot2controller.v1.Get_CameraPicture_Responses) {}
    
}

// ----------------- Data Type definitions -----------------


// ------ Command Parameter and Response definitions -------
message UploadProtocol_Parameters {
    sila2.org.silastandard.String ProtocolSourcePath = 1;
}

message UploadProtocol_Responses {
    
}

message RemoveProtocol_Parameters {
    sila2.org.silastandard.String ProtocolFile = 1;
}

message RemoveProtocol_Responses {
    
}

message RunProtocol_Parameters {
    sila2.org.silastandard.String ProtocolFile = 1;
    sila2.org.silastandard.Boolean IsSimulating = 2;
}

message RunProtocol_Responses {
    sila2.org.silastandard.Integer ReturnValue = 1;
}
message Get_Connection_Parameters {
    // Empty message
}

message Get_Connection_Responses {
    sila2.org.silastandard.String Connection = 1;
}

message Get_AvailableProtocols_Parameters {
    // Empty message
}

message Get_AvailableProtocols_Responses {
    repeated sila2.org.silastandard.String AvailableProtocols = 1;
}

message Get_CameraPicture_Parameters {
    // Empty message
}

message Get_CameraPicture_Responses {
    message CameraPicture_Struct {
        // Image Data
        //   The *.jpg image.
        // Constrained type, not reflected in protocol buffers
        sila2.org.silastandard.Binary ImageData = 1;
        // Image Timestamp
        //   The timestamp when the image was taken.
        sila2.org.silastandard.Timestamp ImageTimestamp = 2;
    }
    sila2.de.fau.dispensing.ot2controller.v1.Get_CameraPicture_Responses.CameraPicture_Struct CameraPicture = 1;
}

// ----------------- Metadata Definitions ------------------
